check_diff/src/lib.rs

```diff
@@ -56,3 +136,139 @@ pub fn change_directory_to_path(dest: &Path) -> io::Result<()> {
     );
     return Ok(());
 }
+
+pub fn get_ld_lib_path() -> Result<String, CheckDiffError> {
+    let Ok(command) = Command::new("rustc").args(["--print", "sysroot"]).output() else {
+        return Err(CheckDiffError::FailedCommand(
+            "Error getting sysroot".to_string(),
+        ));
+    };
+
+    let Ok(sysroot) = String::from_utf8(command.stdout) else {
+        return Err(CheckDiffError::FailedUtf8(
+            "Error converting sysroot to string".to_string(),
+        ));
+    };
+
+    let ld_lib_path = format!("{}/lib", sysroot.trim_end());
+    return Ok(ld_lib_path);
+}
+
+pub fn get_cargo_version() -> Result<String, CheckDiffError> {
+    let Ok(command) = Command::new("cargo").args(["--version"]).output() else {
+        return Err(CheckDiffError::FailedCargoVersion(
+            "Failed to obtain cargo version".to_string(),
+        ));
+    };
+
+    let Ok(cargo_version) = String::from_utf8(command.stdout) else {
+        return Err(CheckDiffError::FailedUtf8(
+            "Error converting cargo version to string".to_string(),
+        ));
+    };
+
+    return Ok(cargo_version);
+}
+
+pub fn copy_src_to_dst(src: &str, dst: &str) -> Result<(), CheckDiffError> {
+    let Ok(_) = Command::new("cp").args([src, dst]).output() else {
+        return Err(CheckDiffError::FailedCopy(
+            "Error copying rustfmt release to destination".to_string(),
+        ));
+    };
+    return Ok(());
+}
+
+pub fn build_rsfmt_from_src(ld_lib_path: &String) -> Result<(), CheckDiffError> {
+    let Ok(_) = Command::new("cargo")
+        .env("LD_LIB_PATH", ld_lib_path)
+        .args(["build", "-q", "--release", "--bin", "rustfmt"])
+        .output()
+    else {
+        return Err(CheckDiffError::FailedSourceBuild(
+            "Error building rustfmt from source".to_string(),
+        ));
+    };
+    return Ok(());
+}
+
+pub fn get_binary_version(binary: String) -> Result<String, CheckDiffError> {
+    let Ok(command) = Command::new(binary.as_str()).args(["--version"]).output() else {
+        return Err(CheckDiffError::FailedVersioning(format!(
+            "Failed to get version for {}",
+            binary
+        )));
```

Comment:
I don't think we need to get extremely granular with our errors. We can just treat this as a `FailedCommand` error.